### DISCUSSION TITLE
How to use both Whatsapp and Telegram?

### DISCUSSION THREAD

**Discussion by HaoZeke** [d99kris/nchat#102] [Q&A]
Hi! Sorry if this is trivial. Is the best way to use multiple protocols to run two instances with different `conf` directories? Or is there a single client multiple messaging backend option of sorts?

P.S. Fantastic project by the way! It's a lifesaver on Wayland distros

**Comment by d99kris** [comment]
Hi @HaoZeke - that is a good question.

**Short answer:** 
Yes, the best way is to run two instances with different conf dirs.

**Long answer:**
So when `nchat v2` was developed the code was designed to cater for multiple protocols in a single session, but the functionality was never really tested out (the whatsapp library at the time was quite unstable). So the multi-protocol functionality ended up disabled by default and I haven't tested it myself after I released `v2` back in 2021 . You _can_ try it out by building `nchat` with something like:
```
mkdir -p build && cd build && cmake -DHAS_MULTIPROTOCOL=ON .. && make -s && sudo make install
```
Then you'd need to set up one protocol at a time, i.e. running `nchat -s` one time for whatsapp and one time for telegram. But it's not officially supported at this point, so any bugs found may or may not get fixed.

**Reply by HaoZeke** [reply]
Thanks, that makes sense, will stick with config folders unless I have time to also contribute for now :)

**Reply by d99kris** [reply]
Hi again @HaoZeke - I finally got around to take a look at this. Support for multiple protocols was added now, see #140.

**Reply by HaoZeke** [reply]
Awesome! Will try it out ASAP.

**Comment by Maverick990** [comment]
Initially, when developing nchat v2, the code was designed to accommodate multiple protocols within a single session. However, due to the instability of the [WhatsApp](https://apkwa.net/whatsapp-plus/) library at that time, the multi-protocol functionality was disabled by default.

**Reply by d99kris** [reply]
I'm not sure what the purpose is with posting this answer? Isn't this simply a rehash of my answer above?

**Comment by HaoZeke** [comment]
Closing the discussion to prevent possible spam :)
